### PR TITLE
Refactoring and Optimization

### PR DESCRIPTION
**Pull request by albernsrya**
Refactoring process tends to yield code which is far easier to maintain and extend in the long run. the benefits of refactoring by applying combinations of refactoring methods on an unoptimized code and
identify which combination of refactoring methods results in better optimization of code

### CODE REVIEW

**Comment by StillerPatrick** [comment]
Hello @albernsrya thanks for you contribution. But there are very torch specific things which are changed by your reformation like the += in the PINN loss which will lead to convergence issues.  So at the moment I will not accept your contribution.

File: PINNFramework/PINN.py [hunk 1]
```diff
@@ -208,30 +197,38 @@ def pinn_loss(self, training_data):
...
             # initial condition loss
-            if len(training_data["Initial_Condition"]) == 2:
-                pinn_loss = pinn_loss + self.initial_condition(training_data["Initial_Condition"][0][0].type(self.dtype),
-                                                               self.model,
-                                                               training_data["Initial_Condition"][1][0].type(self.dtype))
-            else:
-                raise ValueError("Training Data for initial condition is a tuple (x,y) with x the  input coordinates"
-                                 " and ground truth values y")
+        if len(training_data["Initial_Condition"]) == 2:
+            pinn_loss += self.initial_condition(
```
Comment (StillerPatrick) [inline PINNFramework/PINN.py:206]: This is optimized by your automatic refactoring which will lead to bad behavior in optimizing the PINN.

File: PINNFramework/PINN.py [hunk 2]
```diff
@@ -208,30 +197,38 @@ def pinn_loss(self, training_data):
...
+                training_data["PDE"][0].type(self.dtype), self.model
+            )
+
         else:
             raise ValueError("Training Data for PDE data is a single tensor consists of residual points ")
         if not self.is_hpm:
             if isinstance(self.boundary_condition, list):
                 for bc in self.boundary_condition:
-                    pinn_loss = pinn_loss + self.calculate_boundary_condition(bc, training_data[bc.name])
+                    pinn_loss += self.calculate_boundary_condition(bc, training_data[bc.name])
```
Comment (StillerPatrick) [inline PINNFramework/PINN.py:225]: Here the same issue

File: PINNFramework/PINN.py [hunk 3]
```diff
@@ -208,30 +197,38 @@ def pinn_loss(self, training_data):
...
             else:
-                pinn_loss = pinn_loss + self.calculate_boundary_condition(self.boundary_condition,
-                                                                          training_data[self.boundary_condition.name])
+                pinn_loss += self.calculate_boundary_condition(
```
Comment (StillerPatrick) [inline PINNFramework/PINN.py:227]: and here.

**Comment by StillerPatrick** [comment]
Furthermore the 2D Schrödinger code is not in the framework style and will be updated if its in neural solvers fashion.

**Review by StillerPatrick — CHANGES_REQUESTED** [review]
